tests/ttnn/unit_tests/operations/test_maxpool2d.py

```diff
@@ -212,3 +208,254 @@ def test_run_max_pool(
+            1,
+            1,
+            {
+                "num_cores_nhw": 4,
+            },
+            False,
+        ),
+        (
+            1,
+            32,
+            17,
+            17,
+            3,
+            3,
+            2,
+            2,
+            1,
+            1,
+            {
+                "num_cores_nhw": 2,
+            },
+            False,
+        ),
+        (
+            2,
+            32,
+            16,
+            16,
+            3,
+            3,
+            1,
+            1,
+            1,
+            1,
+            {
+                "num_cores_nhw": 3,
+            },
+            False,
+        ),
+        (
+            2,
+            32,
+            23,
+            23,
+            3,
+            3,
+            2,
+            2,
+            1,
+            1,
+            {
+                "num_cores_nhw": 3,
+            },
+            False,
+        ),
+        (
+            1,
+            32,
+            23,
+            23,
+            3,
+            3,
+            1,
+            1,
+            1,
+            1,
+            {
+                "num_cores_nhw": 10,
+            },
+            True,
+        ),
+    ),
+)
+@pytest.mark.parametrize("dtype", [ttnn.bfloat16])
+def test_pool_core_nondivis(
+    device,
+    use_program_cache,
+    batch_size,
+    input_channels,
+    input_height,
+    input_width,
+    filter_height,
+    filter_width,
+    stride_h,
+    stride_w,
+    pad_h,
+    pad_w,
+    config_override,
+    xfail,
+    dtype,
+):
+    if xfail:
+        pytest.xfail()
+
+    act_shape = (batch_size, input_channels, input_height, input_width)
+    in_n, in_c, in_h, in_w = batch_size, input_channels, input_height, input_width
+    kernel_h, kernel_w = filter_height, filter_width
+    dilation_h, dilation_w = 1, 1
+
+    if 2 * pad_h > kernel_h or 2 * pad_w > kernel_w:
+        pytest.skip("Invalid case")
+
+    if (kernel_h == 3 and pad_h != 1) or (kernel_h == 2 and pad_h != 0):
+        pytest.skip("kernel size and padding combination not supported")
+
+    out_h = math.floor((in_h + 2 * pad_h - (dilation_h * kernel_h - 1) - 1) / stride_h) + 1
+    out_w = math.floor((in_w + 2 * pad_w - (dilation_w * kernel_w - 1) - 1) / stride_w) + 1
+    if in_c % 16 != 0:
+        pytest.skip("Current maxpool writer needs nchannels to be multiple of 16!")
+
+    if in_c == 16 and dtype == ttnn.bfloat8_b and in_n * in_h * in_w > 600000:
+        pytest.skip("This case runs out of memory on Grayskull")
+
+    if in_n > 16 and in_c > 64 and dtype == ttnn.bfloat8_b and is_wormhole_b0():
+        pytest.skip("This case runs out of memory on Wormhole b0")
+
+    torch.manual_seed(0)
+    torch.set_printoptions(precision=3, sci_mode=False, linewidth=500, threshold=10000, edgeitems=32)
+
+    ## construct the tensor in NCHW shape
+    act = torch.randn(act_shape, dtype=torch.bfloat16)
+    # act = torch.zeros(act_shape, dtype=torch.bfloat16)
```

Comment:
remove?

Reply:
I like to keep these as they help in debugs :)

Reply:
I'd encourage you to turn this into a testing util function then. Someone might delete this code in another PR as dead.